apps/hyperdrive-trading/src/ui/hyperdrive/longs/CloseLongModalButton/CloseLongModalButton.tsx

```
@@ -45,3 +97,26 @@ export function CloseLongModalButton({
     />
   );
 }
+function getSubHeadingLabel(
+  baseToken: TokenConfig<EmptyExtensions>,
+  hyperdrive: HyperdriveConfig,
+  sharesToken: TokenConfig<YieldSourceExtensions>,
+) {
+  if (
+    hyperdrive.withdrawOptions.isBaseTokenWithdrawalEnabled &&
+    hyperdrive.withdrawOptions.isShareTokenWithdrawalEnabled
+  ) {
+    return `Redeem your hy${baseToken.symbol} for ${baseToken.symbol} or ${sharesToken.symbol}`;
+  }
+
+  if (hyperdrive.withdrawOptions.isBaseTokenWithdrawalEnabled) {
+    return `Redeem your hy${baseToken.symbol} for ${baseToken.symbol}`;
+  }
+
+  if (hyperdrive.withdrawOptions.isShareTokenWithdrawalEnabled) {
+    return `Redeem your hy${baseToken.symbol} for ${sharesToken.symbol}`;
+  }
+
+  // This should never happen and is just to prevent typescript from complaining
+  return "";
```

If you throw an error it satisfies typescript:
`throw new Error("No withdrawal options are enabled");`
If this should never happen we should throw an error anyways.

---

I like that